packages/mui-material/src/Slider/Slider.d.ts

```diff
@@ -1,23 +1,26 @@
 import * as React from 'react';
-import {
-  ExtendSliderUnstyledTypeMap,
-  ExtendSliderUnstyled,
-  SliderUnstyledTypeMap,
-  SliderValueLabelProps,
-} from '@mui/base/SliderUnstyled';
+import { SlotComponentProps } from '@mui/base';
+import { SliderUnstyledTypeMap } from '@mui/base/SliderUnstyled';
 import { SxProps } from '@mui/system';
 import { OverridableStringUnion } from '@mui/types';
 import { Theme } from '../styles';
-import { OverrideProps } from '../OverridableComponent';
+import { OverrideProps, OverridableComponent } from '../OverridableComponent';
+import SliderValueLabelComponent from './SliderValueLabel';
+import { SliderValueLabelProps } from './SliderValueLabel.types';
 
 export interface SliderPropsColorOverrides {}
 
 export interface SliderPropsSizeOverrides {}
 
-export type SliderTypeMap<
-  D extends React.ElementType = 'span',
-  P = {},
-> = ExtendSliderUnstyledTypeMap<{
+export interface SliderComponentsPropsOverrides {}
+
+export interface SliderOwnerState extends SliderProps {
+  dragging: boolean;
+  marked: boolean;
+  focusedThumbIndex: number;
+}
+
+export interface SliderTypeMap<D extends React.ElementType = 'span', P = {}> {
```

Comment:
I had to update the types for Material UI Slider to avoid the change in prop type for `slotProps.valueLabel` slot as mentioned in comment https://github.com/mui/material-ui/pull/35479#discussion_r1059388313.

The SliderValueLabelUnstyled component is removed in this PR, and the value label props depended on it `(typeof SliderValueLabelUnstyled)`. Now it depends on the Material UI SliderValueLabel. `(typeof SliderValueLabel)`.

This is actually an improvement in types, particularly for `componentsProps` and `slotProps` props. There should not be any breaking changes. I have also added the TypeScript tests below.